### PR TITLE
Domaintools Phisheye plugin - add enum parameter

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
## Proposed Changes
Domaintools Phisheye plugin - add enum parameter

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "error": "('action input JSON was invalid', \u003cValidationError: '0 is not one of [1, 2, 3, 4, 5, 6, 7]'\u003e)",
    "log": "Connect: Connecting..\n('action input JSON was invalid', \u003cValidationError: '0 is not one of [1, 2, 3, 4, 5, 6, 7]'\u003e)\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 380, in start_step\n    step.input.validate(params)\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/variables.py\", line 24, in validate\n    validate(parameters, self.schema)\n  File \"/usr/local/lib/python3.7/site-packages/jsonschema-2.3.0-py3.7.egg/jsonschema/validators.py\", line 432, in validate\n    cls(schema, *args, **kwargs).validate(instance)\n  File \"/usr/local/lib/python3.7/site-packages/jsonschema-2.3.0-py3.7.egg/jsonschema/validators.py\", line 117, in validate\n    raise error\njsonschema.exceptions.ValidationError: 0 is not one of [1, 2, 3, 4, 5, 6, 7]\n\nFailed validating 'enum' in schema['properties']['days_back']:\n    {'description': 'Use this parameter in exceptional circumstances where '\n                    'you need to find domains up to seven days prior to '\n                    'the current date. Set the value to an integer in the '\n                    'range of 1-7',\n     'enum': [1, 2, 3, 4, 5, 6, 7],\n     'order': 2,\n     'title': 'Days Back',\n     'type': 'integer'}\n\nOn instance['days_back']:\n    0\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 311, in handle_step\n    output = self.start_step(input_message['body'], 'action', logger, log_stream, is_test, is_debug)\n  File \"/usr/local/lib/python3.7/site-packages/komand-1.0.1-py3.7.egg/komand/plugin.py\", line 387, in start_step\n    raise ClientException('{} input JSON was invalid'.format(step_key), e)\nkomand.exceptions.ClientException: ('action input JSON was invalid', \u003cValidationError: '0 is not one of [1, 2, 3, 4, 5, 6, 7]'\u003e)\n",
    "meta": {},
    "status": "error"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
Failed test - before changes
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting..\nrapid7/DomainTools PhishEye:1.0.0. Step name: domain_list\n",
    "meta": {},
    "output": {
      "date": "2020-09-24",
      "domains": [
        {
          "created_date": "2020-09-22",
          "domain": "rapid.surf",
          "ip_addresses": [
            {
              "country_code": "DE",
              "ip": "78.46.179.15"
            },
            {
              "country_code": "DE",
              "ip": "78.46.179.15"
            }
          ],
          "name_servers": [
            "ns3.second-ns.de"
          ],
          "registrant_email": "Please query the RDDS service of the Registrar of Record identified in this output for information on how to contact the Registrant, Admin, or Tech contact of the queried domain name.",
          "registrar_name": "Key-Systems, LLC",
          "risk_score": 10,
          "tld": "surf"
        }
      ],
      "term": "rapid7"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/domaintools_phisheye:1.0.0 --debug run < tests/domain_list.json
</summary>
</details>

<details>

```

[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 109441.933ms
```

<summary>
icon-validate --all .
</summary>
</details>

<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 966.0849999999999ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
./icon_domaintools_phisheye/connection/connection.py:26:9: F841 local variable 'e' is assigned to but never used
[FAIL] Fails flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.3
Run started:2020-09-24 21:17:39.655554

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 350
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

[*] Checking for typos with Misspell...
[SUCCESS] Passes Misspell check

```

<summary>
make validate
</summary>
</details>

### CODE REVIEW

File: domaintools_phisheye/help.md [hunk 1]
```diff
@@ -177,6 +177,7 @@ If a term is searched for in the Domain List action but not monitored in PhishEy
 
 # Version History
 
+* 1.0.1 - Add `0` parameter to enum in Domain List action
```
Comment (mrinehart-r7) [inline domaintools_phisheye/help.md:180]:
```suggestion
* 1.0.1 - Add `0` parameter to Days Back input in Domain List action to get current day results
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by ncramer-r7 — APPROVED** [review]

**Review by jschipp-r7 — APPROVED** [review]
